main.py

```diff
@@ -63,6 +64,8 @@ def setup(self):
         # Every create/update/delete must be synced to mongodb.
         self.circuits = {}
 
+        self._intf_events = defaultdict(dict)
+        self.lock_interfaces = defaultdict(Lock)
```

Comment:
Same comment regarding the `_` prefix since this is private lock related stuff.

Reply:
Fixed.